### PR TITLE
OPHJOD-968: Update job opportunity page to match latest design

### PR DESCRIPTION
**Pull request by juhaniko**
## Description

* Updated Työmahdollisuus page to match latest design https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=4984-9497&t=tnmpiu2CqY5JbmZO-4
  * Removed accordions
  * Removed tabs and competences page
  * Updated font sizes, gaps and margins
  * Added the toolbar
* Extracted `ActionButton`,  `FavoriteToggle` and `LoginModal` into reusable components away from `OpportunityCard`.
* I did not find the exact icon for "Luo polku", so I used the closest matching one from Material Icons and rotated it to match design
  * 
image vs 
image

## Related JIRA ticket

https://jira.eduuni.fi/browse/OPHJOD-968

image

### CODE REVIEW

File: src/routes/JobOpportunity/Overview.tsx [hunk 1]
```diff
@@ -86,137 +111,103 @@ const Overview = () => {
         </SimpleNavigationList>
       }
     >
+      {loginModalOpen && <LoginModal onClose={() => setLoginModalOpen(false)} isOpen={loginModalOpen} />}
       <Title value={title ?? ''} />
       <h1 className="mb-5 text-heading-2 sm:text-heading-1">{title}</h1>
-      <Tabs />
-      <div className="flex flex-col gap-11">
+      <div className="flex flex-row gap-7 my-8">
```
Comment (sauanto) [inline src/routes/JobOpportunity/Overview.tsx:117]: Hide action buttons in print view
```suggestion
      <div className="flex flex-row gap-7 my-8 print:hidden">
```

File: src/routes/JobOpportunity/Overview.tsx [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
 import { getLocalizedText } from '@/utils';
-import { Accordion } from '@jod/design-system';
 import React from 'react';
 import { useTranslation } from 'react-i18next';
+import { MdCompareArrows, MdOutlinePrint, MdOutlineRoute, MdOutlineShare } from 'react-icons/md';
 import { useOutletContext } from 'react-router-dom';
-import Tabs from './Tabs';
 import { LoaderData } from './loader';
 
 const Overview = () => {
```
Comment (sauanto) [inline src/routes/JobOpportunity/Overview.tsx:21]: Remove the sub-route from the pathname and rename **Overview** to **JobOpportunity**.

**Review by sauanto — CHANGES_REQUESTED** [review]

File: src/routes/JobOpportunity/JobOpportunity.tsx [hunk 3]
```diff
@@ -1,9 +1,217 @@
...
+          <h2 id={t('job-opportunity.labour-market-picture.title')} className="text-heading-2 scroll-mt-[96px]">
+            {t('job-opportunity.labour-market-picture.title')}
+          </h2>
+          <p className="text-body-md font-arial mb-6 mt-4">{t('job-opportunity.labour-market-picture.description')}</p>
+          <div className="bg-todo h-[380px]" />
+        </div>
+        <div>
+          <h2 id={t('job-opportunity.salary-trends.title')} className="text-heading-2 scroll-mt-[96px]">
+            {t('job-opportunity.salary-trends.title')}
+          </h2>
```
Comment (ketsappi) [inline src/routes/JobOpportunity/JobOpportunity.tsx:184]: It came as an idea to have a component for this as this same pattern repeats so many time.

```tsx
const H2Scroll = ({ title }: { title: string }) => {
  return (
    <h2 id={title} className="text-heading-2 scroll-mt-[96px]">
      {title}
    </h2>
  );
};
```

Usage:
```tsx
<H2Scroll title={t('job-opportunity.key-figures.title')} />
```

File: src/routes/JobOpportunity/JobOpportunity.tsx [hunk 4]
```diff
@@ -1,9 +1,217 @@
...
+          <p className="text-body-md font-arial mb-6 mt-4">{t('job-opportunity.employment-trends.description')}</p>
+          <div className="bg-todo h-[380px]" />
+        </div>
+        <div>
+          <h2 id={t('job-opportunity.related-jobs.title')} className="text-heading-2 scroll-mt-[96px]">
+            {t('job-opportunity.related-jobs.title')}
+          </h2>
+          <p className="text-body-md font-arial mb-6 mt-4">{t('job-opportunity.related-jobs.description')}</p>
+          <div className="bg-todo h-[380px] mb-8" />
+        </div>
```
Comment (ketsappi) [inline src/routes/JobOpportunity/JobOpportunity.tsx:211]: And while this is just a todo block and such, we still could improve readability for the time being by using component:

```tsx
const WipBlock = ({ label }: { label: string }) => {
  return (
    <>
      <p className="text-body-md font-arial mb-6 mt-4">{label}</p>
      <div className="bg-todo h-[380px]" />
    </>
  );
};
```

Usage (together with that H2Scroll ):

```tsx
<div className="mb-8">
  <H2Scroll title={t('job-opportunity.related-jobs.title')} />
  <WipBlock label={t('job-opportunity.related-jobs.description')} />
</div>
```

File: src/routes/JobOpportunity/JobOpportunity.tsx (hunk 3 above)
Comment (sauanto) [reply src/routes/JobOpportunity/JobOpportunity.tsx:184]: The heading level could be a prop. This way, it would be easier to use in different situations.

**Review by sauanto — APPROVED** [review]

File: src/routes/JobOpportunity/JobOpportunity.tsx (hunk 4 above)
Comment (juhaniko) [reply src/routes/JobOpportunity/JobOpportunity.tsx]: A solid idea, but I feel it's not worth the effort to start wrapping todo-stuff in todo-components for the sake of readability. Maybe if there were more of these todo blocks.

**Pull request merged**